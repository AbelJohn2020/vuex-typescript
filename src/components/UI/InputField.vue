<template>
  <div class="form-control" :class="{invalid: validation}">
    <label :for="forLabel" id="for-label">
      {{ forLabel }}
      <slot name="valid"></slot>
    </label>
    <slot></slot>
    <p v-if="validation">invalid field</p>
  </div>
</template>

<script>
export default {
    props: [
        'forLabel',
        'validation',
    ]
}
</script>

<style>
.form-control {
  margin: 1rem 0;
  width: 100%;
  display: flex;
  flex-direction: column;
}

#for-label {
  display: flex;
}

label {
    text-transform: capitalize;
}

label, p {
  box-sizing: border-box;
  font-weight: bold;
  text-align: start;
}

p {
    text-transform: capitalize;
    padding: 0;
    margin: 0.4rem 0 0 0;
}

.form-control.invalid label,
.form-control.invalid p {
  color: red;
}
</style>