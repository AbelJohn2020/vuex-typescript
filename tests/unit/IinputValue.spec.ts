import { mount } from '@vue/test-utils';
import InputField from '../../src/components/UI/InputField.vue';

describe('Input field', () => { 
    const wrapper = mount(InputField, {
        slots: {
            default: `
                <input placeholder="Name" id="name" type="text" v-model="valueName"/>
            `,
        },
        props: {
            forLabel: 'name',
            validation: false
        },
        data() {
            return {
                valueName: '',
            };
        },
    });

    const inputValue = wrapper.find('input[type="text"]');


    test('The default Slot in InputField will be Abel Antonio', async () => { 

        await inputValue.setValue('Abel Antonio');

        expect((inputValue.element as HTMLInputElement).value).toBe('Abel Antonio');
    });
 }) 