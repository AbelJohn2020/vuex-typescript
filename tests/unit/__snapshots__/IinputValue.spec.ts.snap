// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input field Should return Invalid Field 1`] = `
<div class="form-control invalid"><label for="name" id="for-label">name </label><input placeholder="Name" id="name" type="text">
  <p>invalid field</p>
</div>
`;

exports[`Input field should return Valid Field 1`] = `
<div class="form-control"><label for="name" id="for-label">name <p>valid field</p></label><input placeholder="Name" id="name" type="text">
  <!--v-if-->
</div>
`;
